### DISCUSSION TITLE
Ideas for distributing emacs juvix-mode

### DISCUSSION THREAD

**Discussion by mariari** [anoma/juvix#1322] [General]
Throughout my emacs config, I am able to load a repo's emacs mode easily enough. For example, for `poplog` I have the following in my config

```elisp
(straight-use-package 'use-package)
(setq straight-use-package-by-default t)
(setq use-package-always-defer t)

(use-package pop-mode
  :straight (:host github :repo "mariari/pop-mode"))
(use-package inferior-pop-mode
  :straight (:host github :repo "mariari/pop-mode"))
(use-package pop-help-mode
  :straight (:host github :repo "mariari/pop-mode"))
```

where I am able to load the modes into my emacs

https://github.com/mariari/pop-mode

however, when I try the same with minijuvix it seems not to work

```elisp
(use-package minijuvix-mode
  :straight (:host github :repo "heliaxdev/minijuvix"))
```

to be fair this is probably my fault for not knowing how to index into a directory.

It would be cool if it is already possible, to document it, and if not to make `minijuvix.el` it's own repo

**Comment by janmasrovira** [comment]
This is not the installation method that's suggested in the readme. I am not saying the method in the readme is better, but it's the one that we support now.

**Comment by jonaprieto** [comment]
There is a new repository for the emacs-mode:

https://github.com/anoma/juvix-mode
